File: backend/app/schemas/preferencias.py
from typing import Literal
from pydantic import BaseModel


AtajoBusqueda = Literal["ctrl_k", "ctrl_p", "ctrl_shift_f", "alt_s"]


DEFAULTS: dict = {
    "busqueda_boton_visible": True,
    "busqueda_atajo": "ctrl_k",
    "sidebar_hidden": [],
}


class PreferenciasOut(BaseModel):
    busqueda_boton_visible: bool
    busqueda_atajo: AtajoBusqueda
    sidebar_hidden: list[str]


class PreferenciasUpdate(BaseModel):
    busqueda_boton_visible: bool | None = None
    busqueda_atajo: AtajoBusqueda | None = None
    sidebar_hidden: list[str] | None = None


def merge_with_defaults(stored: dict | None) -> dict:
    return {**DEFAULTS, **(stored or {})}

File: frontend/src/api/preferencias.ts
import { api } from '../lib/api'

export type AtajoBusqueda = 'ctrl_k' | 'ctrl_p' | 'ctrl_shift_f' | 'alt_s'

export interface Preferencias {
  busqueda_boton_visible: boolean
  busqueda_atajo: AtajoBusqueda
  sidebar_hidden: string[]
}

export async function getPreferencias(): Promise<Preferencias> {
  const { data } = await api.get<Preferencias>('/api/users/me/preferencias')
  return data
}

export async function patchPreferencias(patch: Partial<Preferencias>): Promise<Preferencias> {
  const { data } = await api.patch<Preferencias>('/api/users/me/preferencias', patch)
  return data
}
